### PR TITLE
test(spanner): split the backup/restore test cases into two tests

### PR DESCRIPTION
**Pull request by devbww**
Split the `BackupTest.*` test cases between two test binaries (for
each of the generated and hand-written versions) so that each binary
can run in parallel, reducing the overall runtime.

This also requires sub-partioning the test instances so that no two
`RestoreDatabase()` calls can run in the same instance at the same
time.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7370)

### CODE REVIEW

File: google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+  return run_slow_backup_tests;
+}
+
+bool Emulator() {
+  static bool emulator =
+      google::cloud::internal::GetEnv("SPANNER_EMULATOR_HOST").has_value();
+  return emulator;
+}
+
+class BackupTest : public ::google::cloud::testing_util::IntegrationTest {
```
Comment (coryan) [inline google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc:77]: extra nitty nit: the class name does not match the file name.

File: google/cloud/spanner/integration_tests/backup_extra_integration_test.cc [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+  return run_slow_backup_tests;
+}
+
+bool Emulator() {
+  static bool emulator =
+      google::cloud::internal::GetEnv("SPANNER_EMULATOR_HOST").has_value();
+  return emulator;
+}
+
+class BackupTest : public ::google::cloud::testing_util::IntegrationTest {
```
Comment (coryan) [inline google/cloud/spanner/integration_tests/backup_extra_integration_test.cc:73]: extra nitty nit: class name vs. file name.

**Review by coryan — APPROVED** [review]
The bulk of the test code is hard to review, I trust you just moved things.

**Comment by devbww** [comment]
> The bulk of the test code is hard to review,

Yes, unfortunately.  (I don't know whether the review tools use `git diff --find-copies`, or whether it would even help.  I forgot to try it myself.)

> I trust you just moved things.

Yes, modulo the change in instance-partition labels.

File: google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc]: Done.

File: google/cloud/spanner/integration_tests/backup_extra_integration_test.cc (hunk 2 above)
Comment (devbww) [reply google/cloud/spanner/integration_tests/backup_extra_integration_test.cc]: Done.

File: google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc [hunk 3]
```diff
@@ -0,0 +1,484 @@
+// Copyright 2022 Google LLC
```
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc:1]: this is news to me 😉

File: google/cloud/spanner/integration_tests/backup_extra_integration_test.cc [hunk 4]
```diff
@@ -0,0 +1,432 @@
+// Copyright 2022 Google LLC
```
Comment (dbolduc) [inline google/cloud/spanner/integration_tests/backup_extra_integration_test.cc:1]: same

File: google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc (hunk 3 above)
Comment (devbww) [reply google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc:1]: Ergh.

**Pull request merged**